### DISCUSSION TITLE
Best practices for adding a turf.js line to the map?

### DISCUSSION THREAD

**Discussion by crevulus** [uber/react-map-gl#1500] [Q&A]
I'm struggling to add a curved line using turf.js. A lot of the docs around mapbox recommended turf.js, but it doesn't seem to play so nicely with this package... 

I made a stack overflow post about it here: [SO](https://stackoverflow.com/questions/68003679/turn-mapbox-line-into-turf-js-curved-bezier-line-in-react)

Has anyone done something similar before, and can you share any insights?

**Comment by Pessimistress** [comment]
There is nothing wrong in how you are using `<Source>`. If your curve is not rendered then your GeoJSON is invalid. You can test your generated GeoJSON with sites like https://geojson.io

**Reply by crevulus** [reply]
Thanks, that helped, geojson.io in particular. Still struggling to make the line bendy, so will head over to turf.js to ask about why my line is still straight .
